src/hotspot/share/jfr/dcmd/jfrDcmds.cpp

```diff
@@ -223,7 +176,9 @@ void JfrDCmd::invoke(JfrJavaArguments& method, TRAPS) const {
   constructor_args.set_klass(javaClass(), CHECK);
 
   HandleMark hm(THREAD);
-  JNIHandleBlockManager jni_handle_management(THREAD);
+  JNIHandleMark jni_handle_management(THREAD);
+
+  DEBUG_ONLY(JfrJavaSupport::check_java_thread_in_vm(THREAD));
```

Comment:
This method will call into Java below which already checks the thread is in vm so maybe this is not necessary. Even construct_dcmd_instance() has that assert.

Reply:
You're right, it's doubly redundant.  I'll remove it.